src/components/Message/__tests__/MessageCommerce.test.js

```diff
@@ -0,0 +1,432 @@
+    const message = generateAliceMessage({
+      text: 'Some text will go on this message',
+    });
+    const { getByTestId } = await renderMessageCommerce(message);
+    expect(getByTestId(messageCommerceWrapperTestId).className).toContain(
+      '--has-text',
+    );
+  });
+
+  it('should set correct css class modifier if message has not text', async () => {
+    const message = generateAliceMessage({ text: undefined });
+    const { getByTestId } = await renderMessageCommerce(message);
+    expect(getByTestId(messageCommerceWrapperTestId).className).toContain(
+      '--has-no-text',
+    );
+  });
+
+  it('should set correct css class modifier if message has attachments', async () => {
+    const message = generateAliceMessage({ attachments: [pdfAttachment] });
+    const { getByTestId } = await renderMessageCommerce(message);
+    expect(getByTestId(messageCommerceWrapperTestId).className).toContain(
+      '--has-attachment',
+    );
+  });
+
+  it('should set correct css class modifier if message has reactions', async () => {
+    const bobReaction = {
+      type: 'love',
+      user_id: bob.user_id,
+      user: bob,
+      created_at: new Date('2019-12-02T03:34:00'),
+    };
+    const message = generateAliceMessage({
+      latest_reactions: [bobReaction],
+    });
+    const { getByTestId } = await renderMessageCommerce(message);
+    expect(getByTestId(messageCommerceWrapperTestId).className).toContain(
+      '--with-reactions',
+    );
+  });
+
+  it.each([['top'], ['bottom'], ['middle'], ['single']])(
+    "should set correct css class modifier when message's first group style is %s",
+    async (modifier) => {
+      const message = generateAliceMessage();
+      const { getByTestId } = await renderMessageCommerce(message, {
+        groupStyles: [modifier],
+      });
+      expect(getByTestId(messageCommerceWrapperTestId).className).toContain(
+        modifier,
+      );
+    },
+  );
+
+  it.each([
+    ['should display', 'bottom', { shouldDisplay: true }],
+    ['should display', 'single', { shouldDisplay: true }],
+    ['should not display', 'top', { shouldDisplay: false }],
+    ['should not display', 'middle', { shouldDisplay: false }],
+  ])(
+    '%s user avatar when group style is %s',
+    async (_, groupStyle, { shouldDisplay }) => {
+      const message = generateAliceMessage();
+      await renderMessageCommerce(message, {
+        groupStyles: [groupStyle],
+      });
+      if (shouldDisplay) {
+        expect(AvatarMock).toHaveBeenCalledWith(
+          {
+            image: alice.image,
+            size: 32,
+            name: alice.name,
+            onClick: expect.any(Function),
+            onMouseOver: expect.any(Function),
+          },
+          {},
+        );
+      } else {
+        expect(AvatarMock).not.toHaveBeenCalled();
+      }
+    },
+  );
+
+  it('should show the reaction list when message has no text', async () => {
+    const bobReaction = {
```

Comment:
Maybe we can have a generator function for reaction?!! We will need it at other places as well

Reply:
Yeah... I'm using it in multiple places, indeed. I can make a separate PR and replace it everywhere if that's fine (and if no one does it before).

Reply:
I'm going to merge as is to unblock some other PRs and introduce the generators later on a separate PR.